### PR TITLE
Convert ZWave binding to use ESH SerialPort classes

### PR DESCRIPTION
**Pull request by cdjackson**
This modifies the ZWave binding to use the ESH SerialPort classes. This allows serial-over-ip- as per https://github.com/eclipse/smarthome/pull/5560.

I have only tested the binding with a real serial port - not with the IP extension, but that should be handled through the ESH libraries.

Signed-off-by: Chris Jackson <chris@cd-jackson.com>

### CODE REVIEW

**Comment by cdjackson** [comment]
@kaikreuzer is ```org.eclipse.smarthome.io.transport.serial``` still available in the new system?

**Comment by codeaholics** [comment]
Will it have moved to a new (non-ESH) package and bundle name?

**Comment by cdjackson** [comment]
I don't know - hence why I'm asking ;)

However, I'm assuming, maybe incorrectly, that everything hasn't been refactored as the binding is still using all the original package names - if this has changed it would require a major overhaul of every binding.

**Comment by cdjackson** [comment]
@kaikreuzer  or @openhab/core-maintainers any idea about this?

**Comment by openhab-5iver** [comment]
What is the issue? There are [several other bindings](https://github.com/openhab/openhab2-addons/search?q=org.eclipse.smarthome.io.transport.serial&unscoped_q=org.eclipse.smarthome.io.transport.serial) using it, but using Import-Package instead of Require-Bundle in the manifest.

**Comment by cdjackson** [comment]
> What is the issue?

As per the failed CI.

**Comment by openhab-5iver** [comment]
I only get a 404 when clicking Details.

**Comment by cdjackson** [comment]
It probably has expired and needs to be rerun.

**Comment by openhab-5iver** [comment]
```
19:06:18 Java HotSpot(TM) 64-Bit Server VM warning: Insufficient space for shared memory file:
19:06:18     13077

19:06:50  [ERROR] Internal error: java.lang.RuntimeException: Could not instantiate required component: com.google.inject.ProvisionException: Unable to provision, see the following errors:
19:06:50  [ERROR] 
19:06:50  [ERROR] 1) Error notifying InjectionListener org.eclipse.sisu.plexus.PlexusBeanBinder@42ae9e96 of org.eclipse.tycho.p2.resolver.P2DependencyResolver.
19:06:50  [ERROR] Reason: java.lang.RuntimeException: java.io.IOException: Could not create temp dir /tmp/tycho3657383411294228714equinox
19:06:50  [ERROR] while locating org.eclipse.tycho.p2.resolver.P2DependencyResolver
19:06:50  [ERROR] at ClassRealm[extension>org.eclipse.tycho:tycho-maven-plugin:1.3.0, parent: sun.misc.Launcher$AppClassLoader@7852e922] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)
19:06:50  [ERROR] while locating org.eclipse.tycho.core.DependencyResolver annotated with @com.google.inject.name.Named(value=p2)
19:06:50  [ERROR] 
19:06:50  [ERROR] 1 error
19:06:50  [ERROR] role: org.eclipse.tycho.core.DependencyResolver
19:06:50  [ERROR] roleHint: p2
19:06:50  [ERROR] -> [Help 1]
19:06:50  org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Could not instantiate required component
19:06:50      at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:120)
```
It looks like Jenkins is out of space again... https://github.com/openhab/openhab-core/pull/639#issuecomment-470635445.

**Comment by cweitkamp** [comment]
> is `org.eclipse.smarthome.io.transport.serial` still available in the new system?

Yes, it is. Me did not change the namespaces for now. We just moved the code to a different location.

File: META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -49,3 +49,4 @@ Import-Package: com.google.common.collect,
 Service-Component: OSGI-INF/*.xml
 Export-Package: org.openhab.binding.zwave
 Bundle-ActivationPolicy: lazy
+Require-Bundle: org.eclipse.smarthome.io.transport.serial
```
Comment (wborn) [inline META-INF/MANIFEST.MF]: To fix the build just remove this `Require-Bundle` manifest entry. Instead you should add `org.eclipse.smarthome.io.transport.serial` to the `Import-Package` entries.

**Comment by RafalLukawiecki** [comment]
What is the current status of this? Shows as merged but I saw comments on the OpenHAB forum that it has been removed from the binding. Hoping to remote ZWave stick away from OH server, reliability would be important but reading about issues with the socat ZWave solution after connection losses.

**Comment by cdjackson** [comment]
> What is the current status of this? Shows as merged 

It is merged.

**Pull request merged**